### ISSUE TITLE
lockfile crash on high concurrency

### ISSUE THREAD

**Issue by zkat** [zkat/cacache#21]
So it turns out the current usage of `lockfile` implodes when you have a bunch of things trying to grab a lockfile at the same time (even in a relatively short amount of time). It starts spitting out a bunch of `EEXIST` errors like:

```
{ Error: EEXIST: file already exists, open '/path/to.lock'
  errno: -17,
  code: 'EEXIST',
  syscall: 'open',
  path: '/path/to.lock' }
```

Passing config values such as https://github.com/npm/npm/blob/latest/lib/utils/locker.js#L27-L29 (or at least some reasonable defaults) made these errors completely go away when I just tossed the current npm defaults in.

Tests for this one should be fun 😬 . Any tests that are written for this must first fail both locally and on CI (appveyor and travis alike).

**Comment by zkat** [comment]
Update: I published a quick stopgap to this in `cacache@5.0.2` (ed658262aa6f200d65b58e74ce3f7eebf40fdfe4) because I got tired of floating my own patch around. This should still be implemented, with configuration, and actual tests. I still consider it a bug.

**Comment by zkat** [comment]
#56 should address this one if it gets approved. Woohoo.
